lib/Sema/TypeCheckType.cpp

```diff
@@ -2414,20 +2414,21 @@ Type TypeResolver::resolveAttributedType(TypeAttributes &attrs,
       if (!attrs.has(i))
         continue;
 
-      auto diag = diagnoseInvalid(repr, attrs.getLoc(i),
-                                  diag::attribute_requires_function_type,
-                                  TypeAttributes::getAttrName(i));
-
-      // If we see @escaping among the attributes on this type, because it isn't
-      // a function type, we'll remove it.
-      if (i == TAK_escaping) {
+      if (i == TAK_escaping && ty->getOptionalObjectType()) {
+        auto diag = diagnoseInvalid(repr, attrs.getLoc(i),
+                                    diag::escaping_optional_type_argument);
         diag.fixItRemove(getTypeAttrRangeWithAt(Context,
-                                                attrs.getLoc(TAK_escaping)));
-        // Specialize the diagnostic for Optionals.
-        if (ty->getOptionalObjectType()) {
-          diag.flush();
-          diagnoseInvalid(repr, repr->getLoc(),
-                          diag::escaping_optional_type_argument);
+                                              attrs.getLoc(TAK_escaping)));
```

Comment:
Could you align the arguments here as they were?

Reply:
Yeap, of course, done